test/NodeIPCSMSpec.hs

```diff
@@ -164,27 +193,35 @@ nodeIPCSM serverHandles = StateMachine
   where
     -- | Let's handle just Ping/Pong for now.
     mTransitions :: Model r -> Action r -> Response r -> Model r
-    mTransitions model    action      response        = model
+    mTransitions (RunningModel model) action response   = RunningModel $ model
         { inputMessages     = actionToConcrete action       : inputMessages model
         , outputMessages    = responseToConcreate response  : outputMessages model
         }
+    mTransitions _ ShutdownCmd _                        = ShutdownInit
+    mTransitions ShutdownInit _ _                       = ShutdownInit
 
     -- | Preconditions for this model.
     mPreconditions :: Model Symbolic -> Action Symbolic -> Logic
-    mPreconditions model _ = length (inputMessages model) - length (outputMessages model) .== 0
+    mPreconditions (RunningModel model) _   =
+        length (inputMessages model) - length (outputMessages model) .== 0
+    mPreconditions ShutdownInit         _   = Bot
 
     -- | Post conditions for the system.
     mPostconditions :: Model Concrete -> Action Concrete -> Response Concrete -> Logic
-    mPostconditions _   PingCmd         PongResponse          = Top -- valid
-    mPostconditions _   PingCmd         _                     = Bot -- invalid
-    mPostconditions _   QueryPortCmd    (ReplyPortResponse _) = Top -- valid
-    mPostconditions _   QueryPortCmd    _                     = Bot -- invalid
+    mPostconditions _   PingCmd         PongResponse                = Top -- valid
+    mPostconditions _   PingCmd         _                           = Bot -- invalid
+    mPostconditions _   QueryPortCmd    (ReplyPortResponse _)       = Top -- valid
+    mPostconditions _   QueryPortCmd    _                           = Bot -- invalid
+    mPostconditions _   ShutdownCmd     (ShutdownInitiatedResponse) = Top
+    mPostconditions _   ShutdownCmd     _                           = Bot
 
     -- | Generator for symbolic actions.
     mGenerator :: Model Symbolic -> Maybe (Gen (Action Symbolic))
-    mGenerator _ =  Just $ oneof
+    mGenerator ShutdownInit = Nothing
```

Comment:
When you get `ShutdownInit`, you should stop generating any additional commands/actions?